File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = []
        for row in matrix_string.splitlines():
            column = []
            for value in row.split():
                column.append(int(value))
            self.matrix.append(column)

        self.num_rows = len(self.matrix)
        self.num_columns = len(self.matrix[0])

    def row(self, index):
        if index <= 0 or index > self.num_rows:
            raise Exception("Invalid row index")

        return list(self.matrix[index - 1])


    def column(self, index):
        if index <= 0 or index > self.num_columns:
            raise Exception("Invalid column index")

        return [row[index - 1] for row in self.matrix]

File: lua/matrix/matrix.lua
return function(input)
    local matrix = {}

    for line in input:gmatch('[%d ]+') do
        local row = {}

        for value in line:gmatch('%d+') do
            local number = tonumber(value)

            table.insert(row, number)
        end

        table.insert(matrix, row)
    end

    matrix.row = function(index)
        return matrix[index]
    end

    matrix.column = function(index)
        local col = {}

        for i = 1, #matrix do
            table.insert(col, matrix[i][index])
        end

        return col
    end

    return matrix
end
